.github/workflows/uitests.yml

```diff
@@ -0,0 +1,69 @@
+name: Build main target
+
+on:
+  pull_request:
+    paths:
+      - '.github/workflows/**'
+      - NextcloudTalk.xcodeproj/**
+      - NextcloudTalk/**
+      - NotificationServiceExtension/**
+      - ShareExtension/**
+
+    push:
+      branches:
+        - main
+        - master
+        - stable*
+
+permissions:
+  contents: read
+
+jobs:
+  build-and-test:
+    name: Build and Test
+    runs-on: macOS-latest
+    if: github.event.pull_request.draft == false
+
+    strategy:
+      matrix:
+        test-branches: ['stable27', 'master']
+
+    env:
+      WORKSPACE: NextcloudTalk.xcworkspace
+      DESTINATION: platform=iOS Simulator,name=iPhone 14,OS=16.2
+      SCHEME: NextcloudTalk
+      TEST_BRANCH: ${{ matrix.test-branches }}
+      CONTAINER_NAME: nextcloud_test
+      SERVER_PORT: 8080
+
+    steps:
+    - name: Checkout app
+      uses: actions/checkout@v3
+      with:
+        submodules: true
+
+    - name: Create docker test server
+      run: |
+        source ./ci-create-docker-server.sh
+
+    - name: Set up dependencies talk-ios
+      run: |
+        pod install
+
+    - name: Wait for server
+      run: |
+        source ./ci-wait-for-server.sh
+
+    - name: Install talk
+      run: |
+        source ./ci-install-talk.sh             
```

Comment:
It would be great if the scripts that we use in all 3 projects have the same name. This will make things easier for someone who switches.

Reply:
I thought about this as well. But I think the requirements are different for the projects and there might be more project-specific files and also steps being added in the future (for example you probably don't need to install talk currently). And looking into the future the contents might differ as well, so from my POV this shouldn't be a requirement and will only cause troubles in the long run when a certain file is expected to do a certain thing which it doesn't do.

Reply:
Fair enough 👍